### ISSUE TITLE
Skip analysis on documents whose MSBuild item carries ExcludeFromStyleCop=true metadata

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#726]
The original StyleCop would not produce any messages regarding documents that were brought in via MSBuild items that had ExcludeFromStyleCop=true metadata, such as:

```
<Compile Include="..\SomeSharedFile\CanonicalError.cs">
  <Link>Shared\CanonicalError.cs</Link>
  <ExcludeFromStyleCop>true</ExcludeFromStyleCop>
</Compile>
```

StyleCop.Analyzers does not yet honor this metadata.

**Comment by AArnott** [comment]
I'm emailing the Roslyn folks now to ask how an Analyzer can lookup such metadata

**Comment by felickz** [comment]
Might want to add this documentation to [KnownChanges.md ](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/KnownChanges.md)

**Comment by brunocunhasilva** [comment]
I would like to raise attention to this missing feature - I'm considering StyleCopAnalyzers for an **existent solution with a very large codebase** (dozens of projects with thousands of files) and it's not viable to enforce rules on all existent project files - but we would like to enforce them on newly added files.

Not having this feature completely annihilates applying StyleCopAnalysers on this kind of pre-existent projects - the **Why's?** from original StyleCop:
https://stylecop.codeplex.com/wikipage?title=Using%20StyleCop%20on%20Legacy%20Projects

**Comment by AArnott** [comment]
This is a recognized serious concern within the Visual Studio team. Our current solution is to suppress existing issues using a suppressions file or suppression attributes next to the offending code. But we want to improve on this in the future.

**Comment by vweijsters** [comment]
This will be solved when dotnet/roslyn#3705 gets realized.

**Comment by jake-brandt** [comment]
If anyone knows of the MSFT/StyleCop vision to solve this; I am happy to jump in and contribute

**Comment by sharwell** [comment]
Closing as a duplicate of dotnet/roslyn#3705.

**Comment by innomaxx** [comment]
Hello community!

We had started migrating a large project with StyleCop Classic to `StyleCop.Analyzers`. Here are wide use of `ExcludeFromStyleCop` directives on large amount of files for some reasons. These files are not generated, written by engineers, and count of them is so large.

So we need a solution how to exclude them from being analyzed WITHOUT marking them as generated (that's not true).
We tried use `SuppressMessageAttribute`, but it suppresses just specified rule in `checkId` parameter. We need skip file entirely.

@sharwell any solutions?
